README.rst

```diff
@@ -54,16 +54,51 @@ Client(host, port, auto_reconnect)
 
     auto_reconnect - Optional argument to enable/disable automatic reconnect when connection is lost. Defaults to True if not specified.
 
+    user_id - Optional argument to use when server requires authentication - user id (username)
+
+    password  - Optional argument to use when server requires authentication - user password
+
+    use_encryption - Optional boolean argument to set when server uses encryption
```

Comment:
IMO this is not needed in client. Just makes client more complex. Logic is similar to the port number that is also not auto-detected in the python/JS clients. And you probably anyway need to set up the certificates verification to work it securely etc.

Reply:
It is just a weird boolean, I have not seem similar things in REST APIs where one needs to know whether it is http or https.
Maybe instead the first parameter should be URL (something like 'wss://127.0.0.1')? And if protocol is missing, we presume no encryption for backwards compatibility

Reply:
IMO `use_encryption` is quite user-friendly and should be very understandable that you need to add it to get connected to CDP system that also has "Use Encryption" flag set. 
At least to me it seems more user-friendly, than to explain that you have to add "wss://" to the host when you need to connect to CDP system that has "Use Encryption" flag set.

Reply:
The `use_encryption` is a bit ambiguous. At least my first guess was that the client chooses whether it wants that one specific connection to be encrypted. But actually it is the server that made the choice, so only one value is correct. So the boolean is more like `is_server_using_encryption` which I admit is a bit wordy.

Reply:
I chose `use_encryption` to be similar/familiar to the "Use Encryption" setting in server.

Reply:
Ok, next question. If I get the `use_encryption` wrong, do I get a friendly error message?

Because I might not know whether it is supposed to be encrypted or not if somebody else deployed the system. Maybe CDP should print out "StudioAPIServer: listening on wss://192.168.0.18:7690" or "... Encryption is enabled"?

Reply:
StudioAPIServer is currently printing "StudioAPIServer: listening on 127.0.0.1:7689 (TLS)" - we should maybe state something more understandable instead of `(TLS)`
And maybe we should improve our JS/python clients - they are currently not very user-friendly when you misconfigure them - either port is wrong or host is wrong or authentication errors or encryption problems (e.g. certificate problems) - we should decide how and where to show errors and are they customizable or not.